driver_test.go

```diff
@@ -661,3 +661,366 @@ func TestExecContextDml(t *testing.T) {
+			drop:       "DROP TABLE TestTable ",
+		},
+	}
+
+	// Run tests.
+	for _, tc := range tests {
+
+		// Set up, if required
+		if tc.given != "" {
+			_, err = db.ExecContext(ctx, tc.given)
+			if err != nil {
+				t.Errorf("%s: error setting up: %v", tc.name, err)
+			}
+		}
+		if tc.withInsert != "" {
+			_, err = db.ExecContext(ctx, tc.withInsert)
+			if err != nil {
+				t.Errorf("%s: error setting up (insert): %v", tc.name, err)
+			}
+		}
+
+		// Run test.
+		_, err = db.ExecContext(ctx, tc.when)
+		if (err != nil) && (!tc.thenWantError) {
+			t.Errorf("%s: unexpected query error: %v", tc.name, err)
+		}
+		if (err == nil) && (tc.thenWantError) {
+			t.Errorf("%s: expected query error but error was %v", tc.name, err)
+		}
+
+		// Drop table, if created.
+		if tc.drop != "" {
+			_, err = db.ExecContext(ctx, tc.drop)
+			if err != nil {
+				t.Errorf("%s: error dropping table: %v", tc.name, err)
+			}
+		}
+
+	}
+}
+
+func TestExecContextReferentialIntegrity(t *testing.T) {
+
+	// Open db.
+	ctx := context.Background()
+	db, err := sql.Open("spanner", dsn)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+
+	tests := []struct {
+		name             string
+		givenParent      string
+		givenChild       string
+		withInsertParent string
+		withInsertChild  string
+		when             string
+		thenWantError    bool
+		dropParent       string
+		dropChild        string
+	}{
+		// DDL
+		{
+			name: "drop table referential integrity violation no cascade & foreign key",
+			givenParent: `
+			CREATE TABLE ParentNoCascade (
+				id   INT64,
+			)	PRIMARY KEY (id)`,
+			givenChild: `CREATE TABLE ChildNoCascade (
+				id   INT64,
+				parent_id	INT64,
+				CONSTRAINT fk_nc FOREIGN KEY (parent_id) REFERENCES ParentNoCascade (id)
+			)	PRIMARY KEY (id)`,
+			withInsertParent: `INSERT INTO  ParentNoCascade (id) VALUES (1), (2)`,
+			withInsertChild:  `INSERT INTO  ChildNoCascade (id, parent_id) VALUES (2, 1), (4, 2)`,
+			when:             `DROP TABLE ParentNoCascade`,
+			thenWantError:    true,
+			dropParent:       `DROP TABLE ParentNoCascade`,
+			dropChild:        `DROP TABLE ChildNoCascade`,
+		},
+		{
+			name: "drop table referential integrity violation cascade & interleave",
+			givenParent: `
+			CREATE TABLE ParentCascade (
+				parent_id   INT64,
+			)	PRIMARY KEY (parent_id)`,
+			givenChild: 
+			`CREATE TABLE ChildCascade (
+				parent_id	INT64,
+				id   INT64,
+			)	PRIMARY KEY (parent_id, id),
+			INTERLEAVE IN PARENT ParentCascade ON DELETE CASCADE`,
+			withInsertParent: `INSERT INTO  ParentCascade (parent_id) VALUES (1), (2)`,
+			withInsertChild:  `INSERT INTO  ChildCascade (id, parent_id) VALUES (2, 1), (4, 2)`,
+			when:             `DROP TABLE ParentCascade`,
+			thenWantError: true,
+			dropParent:       `DROP TABLE ParentCascade`,
+			dropChild:        `DROP TABLE ChildCascade`,
+		},
+		// DML
+		{
+			name: "deelte row integrity violation no cascade & foreign key",
```

Comment:
```suggestion
			name: "delete row integrity violation no cascade & foreign key",
```

Reply:
fixed, I laughed at this for a solid 5 mins